### PR TITLE
Improve multiple bags fetch.

### PR DESCRIPTION
**Pull request by mshima**
Fixes https://github.com/jhipster/generator-jhipster/issues/13038

Following discussion at https://github.com/jhipster/generator-jhipster/issues/13038.
This PR implements https://vladmihalcea.com/hibernate-multiplebagfetchexception/ by adding functions to fetch `*-to-many` relationships collections.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            .map(this::fetch<%= relationship.relationshipNameCapitalizedPlural %>)
+<%_ } _%>
+            .get();
+    }
+<%_ for (const relationship of relationships.filter(relationship => relationship.bagRelationship)) { _%>
+
+    <%= asEntity(entityClass) %> fetch<%= relationship.relationshipNameCapitalizedPlural %>(<%= asEntity(entityClass) %> result) {
+        return entityManager
+            .createQuery(
+                "select <%= entityInstance %> from <%= asEntity(entityClass) %> <%= entityInstance %> left join fetch <%= entityInstance %>.<%= relationship.relationshipFieldNamePlural %> where <%= entityInstance %> is :<%= entityInstance %>",
```
Comment (Tcharl) [inline generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs:68]: No way to use CriteriaQuery with JPA metamodel objects? I always found it better to have strongly typed elements for queries

**Comment by DanielFran** [comment]
@mshima, can you update based on main branch, please?

File: generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs (hunk 1 above)
Comment (mshima) [reply generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs]: Not sure, but I think the named approach it's simpler and it's the default for our queries.

File: generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import javax.persistence.EntityManager;
+
+import org.hibernate.annotations.QueryHints;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+
+import <%= packageName %>.domain.<%= persistClass %>;
+
```
Comment (DanielFran) [inline generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs:35]: @mshima Can you reference https://vladmihalcea.com/hibernate-multiplebagfetchexception/ in the class comments please?

**Review by DanielFran — APPROVED** [review]

**Pull request merged**